prov/hook/trace/src/hook_trace.c

```diff
@@ -1154,10 +1173,10 @@ trace_av_open(struct fid_domain *domain, struct fi_av_attr *attr,
 	ret = fi_av_open(dom->hdomain, attr, &myav->hav, &myav->av.fid);
 	if (!ret) {
 		FI_TRACE(dom->fabric->hprov, FI_LOG_DOMAIN,
-		         "av %p, type %d rx_bits 0x%x count %zu ep_per_node %zu "
-		         "name %p flags 0x%lx ctx %p\n",
-		         &myav->av, attr->type, attr->rx_ctx_bits, attr->count,
-		         attr->ep_per_node, attr->name, attr->flags, context);
+		         "av %p, context %p\n"
+		         "%s",
+		         &myav->av, context,
+		         fi_tostr(attr, FI_TYPE_AV_ATTR));
```

Comment:
Swap the order of the patches to add the new fi_tostr() support first, then only use those calls.  This avoids adding a change in the same series that a subsequent patch updates.